src/nupic/algorithms/ColumnPooler.cpp

```diff
@@ -0,0 +1,565 @@
+      // distalConnections.compute();
+    }
+  }
+
+
+  // TODO: apply segment overlap threshold
+  void activateProximalDendrites( const SDR &feedForwardInputs,
+                                  vector<Real> &cellExcitements )
+  {
+    // Proximal Feed Forward Excitement
+    rawOverlaps_.assign( proximalConnections.numSegments(), 0.0f );
+    proximalConnections.computeActivity(rawOverlaps_, feedForwardInputs.getSparse());
+
+    // Setup for Boosting
+    const Real denominator = 1.0f / log2( sparsity / proximalSegments );
+    const auto &af = AF->activationFrequency;
+
+    // Process Each Segment of Each Cell
+    for(auto cell = 0u; cell < proximalConnections.numCells(); ++cell) {
+      Real maxOverlap    = -1.0;
+      UInt maxSegment    = -1;
+      // UInt maxRawOverlap = 0u;
+      for(const auto &segment : proximalConnections.segmentsForCell( cell ) ) {
+        const auto raw = rawOverlaps_[segment];
+        // maxRawOverlap = raw > maxRawOverlap ? raw : maxRawOverlap;
+
+        Real overlap = (Real) raw; // Typecase to floating point.
+
+        // Proximal Tie Breaker
+        // NOTE: Apply tiebreakers before boosting, so that boosting is applied
+        //   to the tiebreakers.  This is important so that the tiebreakers don't
+        //   hurt the entropy of the result by biasing some mini-columns to
+        //   activte more often than others.
+        overlap += tieBreaker_[segment];
+
+        // Normalize Proximal Excitement by the number of connected synapses.
+        const auto nConSyns = proximalConnections.dataForSegment( segment ).numConnected;
+        if( nConSyns == 0 )
+          overlap = 1.0f;
+          // overlap = 0.0f;
+        else
+          overlap /= nConSyns;
+
+        // Boosting Function
+        overlap *= log2( af[segment] ) * denominator;
+
+        // Maximum Segment Overlap Becomes Cell Overlap
+        if( overlap > maxOverlap ) {
+          maxOverlap = overlap;
+          maxSegment = segment;
+        }
+      }
+      proximalMaxSegment_[cell] = maxSegment - cell;
+
+      cellExcitements[cell] = maxOverlap;
+      // Apply Stability & Fatigue
+      X_act[cell]   += (1.0f - stability_rate) * (maxOverlap - X_act[cell] - X_inact[cell]);
+      X_inact[cell] += fatigue_rate * (maxOverlap - X_inact[cell]);
+      cellExcitements[cell] = X_act[cell];
+    }
+  }
+
+
+  void activateCells( vector<Real> &overlaps,
+                      // const SDR    &predictiveCells,
+                            SDR    &activeCells)
+  {
+    const UInt inhibitionAreas = activeCells.size / cellsPerInhibitionArea;
+    const UInt numDesired = (UInt) std::round(sparsity * cellsPerInhibitionArea);
+    NTA_CHECK(numDesired > 0) << "Not enough cellsPerInhibitionArea ("
+      << cellsPerInhibitionArea << ") for desired density (" << sparsity << ").";
+
+    // Compare the cell indexes by their overlap.
+    auto compare = [&overlaps](const UInt &a, const UInt &b) -> bool
```

Comment:
I think we've used something similar (the same) elsewhere, if this is a common use-case, would be nice to add the whole "sort cells by overlap" to connections.

Reply:
Connections can't sort by `overlaps`.  The Connections outputs the `rawOverlap_` which is then transformed by tieBreakers_ & boosting into this `overlaps`.